lib/src/main/java/edu/stanford/futuredata/macrobase/operator/Operator.java

```diff
@@ -1,5 +1,13 @@
 package edu.stanford.futuredata.macrobase.operator;
 
+/**
+ * Generic interface for operations. Should not modify input.
```

Comment:
Now that we have a column store, creating a shallow "copy" of a dataframe with extra or missing columns is cheap and makes the code easier to reason about. Adding or removing rows in a column store is going to require a new copy anyway so no loss there either.

Is there a specific scenario you have in mind where this hurts us?